nsxt/data_source_nsxt_policy_host_transport_node_collection_state.go

```diff
@@ -0,0 +1,132 @@
+				Type:         schema.TypeString,
+				Description:  "Path of the site this Transport Node Collection belongs to",
+				Optional:     true,
+				ValidateFunc: validatePolicyPath(),
+			},
+			"aggregate_progress_percentage": {
+				Type:        schema.TypeInt,
+				Description: "Aggregate percentage of compute collection deployment",
+				Computed:    true,
+			},
+			"cluster_level_error": {
+				Type:        schema.TypeString,
+				Description: "Errors which needs cluster level to resolution",
+				Optional:    true,
+				Computed:    true,
+			},
+			"state": {
+				Type:         schema.TypeString,
+				Description:  "Application state of transport node profile on compute collection",
+				Computed:     true,
+				ValidateFunc: validation.StringInSlice(policyHostTransportNodeCollectionState, false),
+			},
+			"validation_errors": {
+				Type:        schema.TypeList,
+				Description: "Errors while applying transport node profile on discovered node",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"discovered_node_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"error_message": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				Optional: true,
+				Computed: true,
+			},
+			"vlcm_transition_error": {
+				Type:        schema.TypeString,
+				Description: "Errors while enabling vLCM on the compute collection",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceNsxtPolicyHostTransportNodeCollectionStateRead(d *schema.ResourceData, m interface{}) error {
+	if isPolicyGlobalManager(m) {
+		return localManagerOnlyError()
+	}
+
+	objID := d.Get("id").(string)
+	objName := d.Get("display_name").(string)
+	objSitePath := d.Get("site_path").(string)
+	// For local manager, if site path is not provided, use default site
+	if objSitePath == "" {
+		objSitePath = defaultSite
+	}
+	if objID == "" && objName == "" {
+		return fmt.Errorf("please specify id or display_name in order to identify Transport Node Collection")
+	}
+
+	connector := getPolicyConnector(m)
+	client := transport_node_collections.NewStateClient(connector)
+	if objID == "" {
+		query := make(map[string]string)
+		globalPolicyEnforcementPointPath := getGlobalPolicyEnforcementPointPath(m, &objSitePath)
+		query["parent_path"] = globalPolicyEnforcementPointPath
+		obj, err := policyDataSourceResourceReadWithValidation(d, connector, getSessionContext(d, m), "HostTransportNodeCollection", query, false)
+		if err != nil {
+			return err
+		}
+		converter := bindings.NewTypeConverter()
+		dataValue, errors := converter.ConvertToGolang(obj, model.HostTransportNodeCollectionBindingType())
+		if len(errors) > 0 {
+			return errors[0]
+		}
+		hostTransportNodeCollectionResource := dataValue.(model.HostTransportNodeCollection)
+		objID = *hostTransportNodeCollectionResource.Id
+	}
+
+	objState, err := client.Get(objSitePath, getPolicyEnforcementPoint(m), objID)
```

Comment:
Would it make sense to move this under `else`? If we already have the object in hand from the search API, no need to pull it again.